pkgs/by-name/no/nordvpn/package.nix

```
@@ -0,0 +1,174 @@
+{
+  appendOverlays,
+  buildGoModule,
+  copyDesktopItems,
+  e2fsprogs,
+  fetchFromGitHub,
+  iproute2,
+  iptables,
+  lib,
+  libxml2_13,
+  makeDesktopItem,
+  makeWrapper,
+  pkg-config,
+  procps,
+  systemdMinimal,
+  wireguard-tools,
+}:
+let
+  tunnelblickSrc = fetchFromGitHub {
+    owner = "Tunnelblick";
+    repo = "Tunnelblick";
+    tag = "v8.0";
+    hash = "sha256-Kj/F7hI6E+giT+4iGDUjXCLgy/6jcohtTWtLXOpZfo0=";
+  };
+
+  patchedPkgs = appendOverlays [
+    (final: prev: {
+      # Nordvpn uses a patched openvpn in order to perform xor obfuscation
+      # https://github.com/NordSecurity/nordvpn-linux/blob/e61430/ci/openvpn/check_dependencies.sh
+      openvpn = prev.openvpn.overrideAttrs (old: {
+        patches =
+          let
+            patchPathFtn = (
+              fname: "${tunnelblickSrc}/third_party/sources/openvpn/openvpn-${old.version}/patches/${fname}"
+            );
+          in
+          (old.patches or [ ])
+          ++ [
+            (patchPathFtn "02-tunnelblick-openvpn_xorpatch-a.diff")
+            (patchPathFtn "03-tunnelblick-openvpn_xorpatch-b.diff")
+            (patchPathFtn "04-tunnelblick-openvpn_xorpatch-c.diff")
+            (patchPathFtn "05-tunnelblick-openvpn_xorpatch-d.diff")
+            (patchPathFtn "06-tunnelblick-openvpn_xorpatch-e.diff")
+          ];
+      });
+    })
+  ];
```

This can just be `overrideAttrs`'d directly -- this adds at least 200 MiB to the Nixpkgs eval(!):
 
```suggestion
  patchedOpenvpn = openvpn.overrideAttrs (old: {
        patches =
          let
            patchPathFtn = (
              fname: "${tunnelblickSrc}/third_party/sources/openvpn/openvpn-${old.version}/patches/${fname}"
            );
          in
          (old.patches or [ ])
          ++ [
            (patchPathFtn "02-tunnelblick-openvpn_xorpatch-a.diff")
            (patchPathFtn "03-tunnelblick-openvpn_xorpatch-b.diff")
            (patchPathFtn "04-tunnelblick-openvpn_xorpatch-c.diff")
            (patchPathFtn "05-tunnelblick-openvpn_xorpatch-d.diff")
            (patchPathFtn "06-tunnelblick-openvpn_xorpatch-e.diff")
          ];
      });
```

---

That looks much better, though I’d like to see someone get to the bottom of the OpenVPN failure first!

---

Actually I spoke too soon, while the build succeeded the `nordvpnd` binary could not find the openvpn executable. Perhaps we can allow the file bloat for now, and I can open another PR once that issue becomes resolved?

---

What do you mean file bloat?

We unfortunately can’t merge this if it doesn’t run, if that’s what you’re saying.

---

You said that even with the above override fixing the build failure, this still caused issues compared to your previous solution; can you push a reproducer of that?

---

They do not result in a similar derivation because one has an openvpn binary while the other does not.

If I apply the patch you suggested at the start of this thread, then that would (surprisingly) result in a build failure.

Regardless neither of them produce a working package (but the PR as it stands works). Please clarify how you would like to proceed. Ty

---

I had forgotten to prepend `old.patches` :weary: 
```
 (old.patches or [ ])
          ++
```
It appears to work as expected now.